src/prefect/environments/storage/docker.py

```diff
@@ -108,7 +102,14 @@ def __init__(
         self.files = files or {}
         self.flows = dict()  # type: Dict[str, str]
         self._flows = dict()  # type: Dict[str, "prefect.core.flow.Flow"]
-        self.base_url = base_url or default_url
+        self.base_url = os.environ.get(
+            "DOCKER_HOST",
+            base_url
+            if base_url
+            else "unix://var/run/docker.sock"
+            if sys.platform != "win32"
+            else "npipe:////./pipe/docker_engine",
+        )
```

Comment:
While this works it is quite confusing to look at lol. Could you keep the original platform check and then update this to do something like 

```
os.environ.get("DOCKER_HOST", base_url if base_url else default_url)
```